stdlib/public/runtime/Metadata.cpp

```diff
@@ -78,12 +96,20 @@ void *MetadataAllocator::alloc(size_t size) {
   // Allocate a new page if we need one.
   if (LLVM_UNLIKELY(((uintptr_t)next & ~pagesizeMask)
                       != (((uintptr_t)end & ~pagesizeMask)))){
+#if defined(_MSC_VER)
+    next = (char *)
+      VirtualAlloc(nullptr, pagesizeMask+1,
```

Comment:
Could you define a wrapper over mmap()/VirtualAlloc() instead of duplicating the logic?